src/database/models/student-grade.ts

```diff
@@ -1,257 +1,278 @@
+    public locked!: boolean;
+    // This is a jsonb field so it could be any (from db)
+    // Submitted in workbook used any so I'm going to keep it consistent here
+    // If this is used for form data we will never know any info about what keys are available
+    // Might make sense to make this an unknown type since I don't think we will ever access the types
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public currentProblemState!: any;
 
-  public lastInfluencingLegalAttemptId!: number | null;
-  public lastInfluencingCreditedAttemptId!: number | null;
-  public lastInfluencingAttemptId!: number | null;
+    public lastInfluencingLegalAttemptId!: number | null;
+    public lastInfluencingCreditedAttemptId!: number | null;
+    public lastInfluencingAttemptId!: number | null;
 
-  public getUser!: BelongsToGetAssociationMixin<User>;
-  public getQuestion!: BelongsToGetAssociationMixin<CourseWWTopicQuestion>;
-  public getWorkbooks!: HasManyGetAssociationsMixin<StudentWorkbook>;
-  public getOverrides!: HasManyGetAssociationsMixin<StudentGradeOverride>;
+    public getUser!: BelongsToGetAssociationMixin<User>;
+    public getQuestion!: BelongsToGetAssociationMixin<CourseWWTopicQuestion>;
+    public getWorkbooks!: HasManyGetAssociationsMixin<StudentWorkbook>;
+    public getStudentGradeInstances!: HasManyGetAssociationsMixin<StudentGradeInstance>;
+    public getOverrides!: HasManyGetAssociationsMixin<StudentGradeOverride>;
 
-  public user!: User;
-  public courseWWTopicQuestion!: CourseWWTopicQuestion;
-  public workbooks?: Array<StudentWorkbook>;
-  public overrides?: Array<StudentGradeOverride>;
+    public readonly user!: User;
+    public readonly courseWWTopicQuestion!: CourseWWTopicQuestion;
+    public readonly workbooks?: Array<StudentWorkbook>;
+    public readonly gradeInstances?: StudentGradeInstance[];
+    public readonly overrides?: Array<StudentGradeOverride>;
 
-  // timestamps!
-  public readonly createdAt!: Date;
-  public readonly updatedAt!: Date;
+    // timestamps!
+    public readonly createdAt!: Date;
+    public readonly updatedAt!: Date;
 
-  static createAssociations(): void {
-    // This is a hack to add the associations later to avoid cyclic dependencies
-    /* eslint-disable @typescript-eslint/no-use-before-define */
-    // // Here we associate which actually populates out pre-declared `association` static and other methods.
-    // User.hasMany(Session, {
-    //   sourceKey: 'id',
-    //   foreignKey: 'user_id',
-    //   as: 'user' // this determines the name in `associations`!
-    // });
+    static getVersion = (obj: StudentGradeInterface, version: StudentGradeInstanceGradeOverridesInterface): StudentGradeInterface => {
+        // Avoid cyclic dependencies
+        // eslint-disable-next-line @typescript-eslint/no-use-before-define
+        return _.assign({}, obj, version); // will override randomSeed, numAttempts, and currentProblemState ** do we want scores too?
```

Comment:
This isn't doing a pick thus it should be override fields like id as well? I would check because this could subtly make some really major bugs

Reply:
Granted it shouldn't be so bad because it returns the stripped model

Reply:
StudentGradeInstanceGradeOverridesInterface only includes `randomSeed` and `currentProblemState`

Reply:
I imitated what you did for overrides elsewhere

Reply:
Gotcha, it blended in, looks good